crates/types/src/bindings.rs

```diff
@@ -12,51 +12,106 @@ sol!(
     "abi/Zenith.json"
 );
 
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    Passage,
+    "abi/Passage.json"
+);
+
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    HostOrders,
+    "abi/HostOrders.json"
+);
+
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    RollupOrders,
+    "abi/RollupOrders.json"
+);
+
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    OrderDestination,
+    "abi/OrderDestination.json"
+);
+
+sol!(
+    #[sol(rpc)]
+    #[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    OrderOrigin,
+    "abi/OrderOrigin.json"
+);
+
 // Zenith types
 impl Copy for Zenith::BlockHeader {}
 impl Copy for Zenith::BlockSubmitted {}
 impl Copy for Zenith::SequencerSet {}
-impl Copy for Zenith::BadSequence {}
 impl Copy for Zenith::BadSignature {}
-impl Copy for Zenith::BlockExpired {}
 impl Copy for Zenith::OneRollupBlockPerHostBlock {}
 impl Copy for Zenith::OnlySequencerAdmin {}
 
+// TODO ensure these are not reimplemented somewhere else
+// impl Copy for Zenith::BadSequence {}
+// impl Copy for Zenith::BlockExpired {}
+
 // Passage types
-impl Copy for Zenith::Enter {}
-impl Copy for Zenith::Withdrawal {}
-impl Copy for Zenith::OnlyWithdrawalAdmin {}
+impl Copy for Passage::Enter {}
+impl Copy for Passage::EnterToken {}
+impl Copy for Passage::EnterConfigured {}
+impl Copy for Passage::Withdrawal {}
+impl Copy for Passage::OnlyTokenAdmin   {}
+
+// TODO these cann't be impl because they are not Copy? How to fix? 
+// impl Copy for Passage::Transact {}
+// impl Copy for Passage::TransactToken {}
+// impl Copy for Passage::DefaultRollupChainId {}                      
+
+// TODO ensure zenith errors are impl if they are used
+// impl Copy for Zenith::ZenithErrors {}
+// impl Clone for Zenith::ZenithErrors {
+//     fn clone(&self) -> Self {
+//         *self
+//     }
+// }
 
-impl Copy for Zenith::ZenithErrors {}
 
-impl Clone for Zenith::ZenithErrors {
+impl Clone for Zenith::ZenithEvents {
```

Comment:
we don't want to interleave passage and zenith

- all zenith blocks
- all passage blocks
- all orders blocks